eng/common/pipelines/templates/steps/git-push-changes.yml

```diff
@@ -41,7 +41,7 @@ steps:
 
 - task: PowerShell@2
   displayName: Push changes
-  condition: and(succeeded(), eq(variables['HasChanges'], 'true'))
+  condition: and(succeeded(), ne(variables['HasChanges'], 'false'))
```

Comment:
Why this change? We want to make sure we don't accidently start pushing these changes unless someone has explicitly set it to true.

Reply:
Several cases here:
1. Step `Check for changes` executes:
* `git diff` success: Then HasChange=true, we run `git push` on purpose. 
* `git diff` fail: Then HasChange=false, we skip `git push` on purpose. 
2. Step `Check for changes` skips, then HasChanges not set, we will run `git push` step anyway:
* If people intended to check in some commits to branch, then they have to literally put PushArgs: -f. (_This is the purpose of the changes. In my case, I fetch some commits in master, and then merge the commits into live branch_)
* If PushArgs not set, then the script will fail. We have to take care of this case which not fail previously.

Reply:
1. That is has intended.
2. If you set the value to skip the check for changes then you are in charge of setting HasChanges to true if there are changes that are interesting. We want to avoid blindly pushing changes in a case where someone doesn't set HasChanges at all. 

As far as I'm aware this logic is working exactly has intended and if there is some scenario where you feel it is working please give me the specific case as I suspect that case needs to be address individually instead of changing the common template logic.

Reply:
Totally agree with you as the changes here might breaking some feature. I removed it from this PR. Will open a new one if it is needed.